src/js/game/Entities/Ghast.js

```diff
@@ -87,26 +123,133 @@ module.exports = class Ghast extends BaseEntity {
     if (this.audioDelay > 0) {
       --this.audioDelay;
     } else {
-    this.audioDelay = 5;
-    let chance = Math.floor(Math.random() * 5);
-    if (chance === 0) {
-      let soundNum = Math.floor(Math.random() * 4);
-      switch (soundNum) {
-        case 0:
-          this.controller.audioPlayer.play("moan2");
-          break;
-        case 1:
-          this.controller.audioPlayer.play("moan3");
-          break;
-        case 2:
-          this.controller.audioPlayer.play("moan6");
-          break;
-        default:
-          this.controller.audioPlayer.play("moan7");
-          break;
+      this.audioDelay = 5;
+      let chance = Math.floor(Math.random() * 5);
+      if (chance === 0) {
+        let soundNum = Math.floor(Math.random() * 4);
+        this.playMoan(soundNum);
+      }
+    }
+  }
+
+  playMoan(number) {
+    switch (number) {
+      case 0:
+        this.controller.audioPlayer.play("moan2");
+        break;
+      case 1:
+        this.controller.audioPlayer.play("moan3");
+        break;
+      case 2:
+        this.controller.audioPlayer.play("moan6");
+        break;
+      default:
+        this.controller.audioPlayer.play("moan7");
+        break;
+    }
+  }
+
+// this.pattern = {type: <type>, specifics: <orientation or starting justification>};
+  fly() {
+    switch (this.details.pattern) {
+      case "box":
+        this.box(this.details.variant);
+        break;
+      case "sinusoidal":
+        this.sinusoidal(this.details.variant);
+        break;
+      default:
+        //do nothing
+    }
+  }
+
+  sinusoidal(variant) {
```

Comment:
Can we use [`Phaser.Easing.Sinusoidal`](https://phaser.io/docs/2.3.0/Phaser.Easing.Sinusoidal.html) instead of rolling our own? 😄

Reply:
Does that do the same thing? My sinusoidal function is a bit tailored to the levelData format (The amplitude of the wave I don't suspect would be an issue, but the length of the path and the refersing behavior along the primary axis is something I don't -imagine- would be build in to a phaser easing function? Maybe though? I'll read the API